waspc/headless-test/start.js

```diff
@@ -0,0 +1,32 @@
+const cp = require('child_process');
+const readline = require('linebyline');
+
+function spawn(name, cmd, args, done) {
+  const spawnOptions = {
+    detached: true,
+  };
+  const proc = cp.spawn(cmd, args, spawnOptions);
+
+  // We close stdin stream on the new process because otherwise the start-app
+  // process hangs.
+  // See https://github.com/wasp-lang/wasp/pull/1218#issuecomment-1599098272.
+  proc.stdin.destroy();
+
+  readline(proc.stdout).on('line', data => {
+    console.log(`\x1b[0m\x1b[33m[${name}][out]\x1b[0m ${data}`);
+  });
+  readline(proc.stderr).on('line', data => {
+    console.log(`\x1b[0m\x1b[33m[${name}][err]\x1b[0m ${data}`);
+  });
+  proc.on('exit', () => {
+    done();
+  });
+}
+
+let numDone = 0;
+const cb = () => {
+  if (numDone == 1) process.exit(0);
```

Comment:
So this ensures that the execution stops after both things have been executed?

I have to admit code confused me a bit, I think it would be cleaner like this:
```
let numDone = 0;
const cb = () => {
  numDone++;
  if (numDone == 2) process.exit(0);
}
```

Even better might be having `spawn` function increase the global counter `numRunning` and then callback reduces that number and does exit when it is 0.